src/schemas.jl

```diff
@@ -251,48 +251,82 @@ otherwise constitutes type piracy and should be avoided.
 accepted_field_type(::SchemaVersion, ::Type{UUID}) = Union{UUID,UInt128}
 accepted_field_type(::SchemaVersion, ::Type{Symbol}) = Union{Symbol,String}
 
+
+
 """
     Legolas.find_violation(ts::Tables.Schema, sv::Legolas.SchemaVersion)
 
-For required field `f::F` of `sv`:
+Return first schema violation; otherwise, return `nothing`. For description of
+schema violations, see [`Legolas.find_violations`](@ref).
 
-- Define `A = Legolas.accepted_field_type(sv, F)`
-- If `f::T` is present in `ts`, ensure that `T <: A` or else immediately return `f::Symbol => T::DataType`.
-- If `f` isn't present in `ts`, ensure that `Missing <: A` or else immediately return `f::Symbol => missing::Missing`.
+See also: [`Legolas.find_violations`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
+"""
+function find_violation(ts::Tables.Schema, sv::SchemaVersion)
```

Comment:
same as https://github.com/beacon-biosignals/Legolas.jl/pull/85/files#r1169390493

but i'd not deprecate this and instead keep the original more optimal short-circuiting implementation

(this method is useful in scenarios where you have small record batches being communicated and you want the common "this batch is valid" case at the communication boundary to be fast)

Reply:
🙃 https://github.com/beacon-biosignals/Legolas.jl/pull/85#discussion_r1166000126 & #87 

I will revert these!

Reply:
done!

Reply:
>this method is useful in scenarios where you have small record batches being communicated and you want the common "this batch is valid" case at the communication boundary to be fast

@jrevels can you elaborate a bit further on this? I'm curious to know what use case this is